File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.go
func minLength(s string) int {
    arr:= []byte{}
    for i:=0;i<len(s);i++{
        if(len(arr)>0){
            if s[i]=='B' && arr[len(arr)-1]=='A'{
                arr=arr[:len(arr)-1]
                continue;
            }else if s[i]=='D' && arr[len(arr)-1]=='C'{
                arr=arr[:len(arr)-1]
                continue;
            }

        }
        arr = append(arr,s[i]);
    }   
    return len(arr); 
}

File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.cpp
class Solution {
public:
    int minLength(string s) {
        stack<char>st;
        for(char c:s){
            if(!st.empty()){
                if(c=='B' && st.top()=='A'){
                    st.pop();
                    continue;
                }
                else if(c=='D' && st.top()=='C'){
                    st.pop();
                    continue;
                }
            }  
            st.push(c);          
        }        
        return st.size();
    }
};